clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -36,4 +46,46 @@ public EcsClusterProvider(Cache cacheView) {
   public Collection<EcsCluster> getAllEcsClusters() {
     return ecsClusterCacheClient.getAll();
   }
+
+  public Collection<Cluster> getAllEcsClustersDescription(String account, String region) {
+    List<String> clusterNames = new ArrayList<>();
+    List<Cluster> clusters = new ArrayList<>();
+    Collection<EcsCluster> ecsClusters = ecsClusterCacheClient.getAll();
+    AmazonECS client = getAmazonEcsClient(account, region);
+    if (ecsClusters.size() > 0 && ecsClusters != null) {
```

Comment:
these 2 conditions should be reversed, otherwise if `ecsClusters` is null then I believe `ecsClusters.size()` will throw an NPE.

Reply:
Added in the new commit.